src/Model/Contact.php

```diff
@@ -2249,12 +2249,12 @@ public static function magicLink($contact_url, $url = '')
 			return $url ?: $contact_url; // Equivalent to: ($url != '') ? $url : $contact_url;
 		}
 
-		$cid = self::getIdForURL($contact_url, 0, true);
```

Comment:
This "getIdForURL" is calling "getProbeDataFromDatabase" if needed. This is triggered by the "true" in that function call.

Reply:
Yep, but if `getProbeDataFromDatabase` returns nothing, a probe is still done. This PR cuts that part completely to improve front-end performance.

Reply:
I understand what he meant, but I'm more confortable changing this one instance that I know is for frontend than the myriad of uses of `Contact:: getIdForURL` in varying contexts.

Reply:
I suggest a change in ```Contact:: getIdForURL``` so the function just returns a ```0``` when the call of ```getProbeDataFromDatabase``` returns an empty value. This is better, since the value ```$no_update``` is exactly built for calls from the frontend - and this in fact is used in the api.php and conversation.php as well. So it would be better changing this here and leaving the rest untouched.

Reply:
If we particularly needed this method, why not, but we don't even need the contact Id specifically. For magic links we just need the public contact info for a given contact URL without probing, and this is exactly what `Contact::getProbeDataFromDatabase` does.

Reply:
But using the other function reduces the number of database calls, since for a repeated call with the same url it will just be a single database call instead of probing several tables - and especially the ```gcontact``` table can be really huge, which is bad for the performance of that call.

Reply:
Let's compare in the "no information for target contact" case that interests us.

### Patched `Contact::getIDForURL` then `Contact::magicLinkbyId`

- `src/Model/Contact.php` line 1245: 1 query
- `src/Model/Contact.php` line 1249: 1 query
- `src/Model/Contact.php` line 1257: 1 query
- `src/Model/Contact.php` line 1286: Call to `Contact::getProbeDataFromDatabase`
(Probe in line 1294 is skipped)
- `src/Model/Contact.php` line 1207: 1 query
- `src/Model/Contact.php` line 1308: Another call to `Contact::getProbeDataFromDatabase` (?!)
- Finally returns 0

### `Contact::getProbeDataFromDatabase` then `Contact:: magicLinkByContact`

- `src/Model/Contact.php` line 2252: Call to `Contact::getProbeDataFromDatabase`
- Returns empty array

As you can see, in the original patched case we would perform `2 x getProbeDataFromDatabase + 4` queries without the probe. Even from a strict performance standpoint my solution is better.

Reply:
```getIDForURL``` is cached, since it automatically creates an entry in the ```contact``` table, when being called with ```uid``` equals to ```0```.

Reply:
Not if `no_update` is set, like in this usage.

Reply:
Possibly I'm too stubborn right now. I guess I need a break.

Reply:
I'm not in the best position to judge this, but please take all the time you want away from Friendica if you feel you need it. You did an amazing job with the ActivityPub implementation and you deserve some rest from Friendica if you feel like you need it.